sdk/python/feast/offline_server.py

```diff
@@ -431,6 +432,20 @@ def get_historical_features(self, command: dict, key: Optional[str] = None):
             # Extract parameters from the internal flights dictionary
             entity_df_value = self.flights[key]
             entity_df = pa.Table.to_pandas(entity_df_value)
+            # Check if this is a mock/empty table (contains only 'key' column)
+            if len(entity_df.columns) == 1 and "key" in entity_df.columns:
+                entity_df = None
+
+        # If no entity_df provided, create minimal one with timestamps
+        if entity_df is None and "start_date" in command and "end_date" in command:
+            # Create minimal entity_df with event_timestamp when using start_date/end_date mode
+            start_date = utils.make_tzaware(
+                datetime.fromisoformat(command["start_date"])
+            )
+            end_date = utils.make_tzaware(datetime.fromisoformat(command["end_date"]))
+            entity_df = pd.DataFrame(
```

Comment:
We should not need to create dummy entity_df at all. We should handle empty(`entity_df=None`) entity_df for and in `offline_store.get_historical_features` which we are returning in this function.